File: frontend/src/utils/engine.js
export const CANONICAL_ENGINE = "mediapipe";

const ENGINE_ALIASES = new Set([
  CANONICAL_ENGINE,
  "opencv",
  "face_recognition",
  "deepface",
  "insightface",
  "arcface"
]);

export function normalizeEngineName(engine) {
  const value = String(engine || "")
    .trim()
    .toLowerCase();

  if (!value || ENGINE_ALIASES.has(value)) {
    return CANONICAL_ENGINE;
  }

  return CANONICAL_ENGINE;
}

export function getEngineLabel(engine) {
  return normalizeEngineName(engine) === CANONICAL_ENGINE
    ? "MediaPipe Face Embeddings"
    : "MediaPipe Face Embeddings";
}

File: python-ai/utils/engine.py
from typing import List


CANONICAL_ENGINE = "mediapipe"
ENGINE_ALIASES = {
    CANONICAL_ENGINE,
    "opencv",
    "face_recognition",
    "deepface",
    "insightface",
    "arcface",
}


def normalize_engine_name(name: str | None) -> str:
    value = (name or "").strip().lower()

    if not value or value in ENGINE_ALIASES:
        return CANONICAL_ENGINE

    return CANONICAL_ENGINE


def get_storage_engine_names(name: str | None) -> List[str]:
    normalized_name = normalize_engine_name(name)

    if normalized_name == CANONICAL_ENGINE:
        return [CANONICAL_ENGINE, "opencv"]

    return [normalized_name]
